addon/PreviewProvider.js

```diff
@@ -209,16 +209,18 @@ PreviewProvider.prototype = {
     dbLinks.forEach(item => existingLinks.set(item.cache_key, item));
     let results = processedLinks.map(site => {
       let enhancedLink = null;
-
-      if (site) {
-        // Find the item in the map and return it if it exists
-        if (existingLinks.has(site.cache_key)) {
-          enhancedLink = Object.assign({}, existingLinks.get(site.cache_key), site);
-        } else {
-          // Add tippy top data, if available
-          enhancedLink = this._tippyTopProvider.processSite(site);
-        }
+      if (!site) {
```

Comment:
If you move this one line up, you can get rid of 

    let enhancedLink = null;

altogether and just add a let to line 215.